WidgetKit/Helpers.swift

```diff
@@ -15,3 +17,33 @@ func linkToContainingApp(_ urlSuffix: String = "", query: String) -> URL {
     let urlString = "\(scheme)://\(query)\(urlSuffix)"
     return URL(string: urlString)!
 }
+
+func getImageForUrl(_ url: URL, completion: @escaping (Image?) -> Void) {
+    let queue = DispatchQueue.global()
+
+    var fetchImageWork: DispatchWorkItem?
+
+    fetchImageWork = DispatchWorkItem {
+        if let data = try? Data(contentsOf: url) {
+            if let image = UIImage(data: data) {
+                DispatchQueue.main.async {
+                    if fetchImageWork?.isCancelled == true { return }
+
+                    completion(Image(uiImage: image))
+                    fetchImageWork = nil
+                }
+            }
+        }
+    }
+
+    queue.async(execute: fetchImageWork!)
```

Comment:
I think this is fine that we are using forced unwrapping but considering there is `if let` and if that fails then we could get nil. Its rare but not impossible. @sblatz thoughts on this? 

```
if let data = try? Data(contentsOf: url) {
```

Reply:
Wrapped it in this :)